src/downloader.cpp

```diff
@@ -209,9 +209,16 @@ bool downloadCanBeReused(const Download& d,
 
 } // unnamed namespace
 
-std::shared_ptr<Download> Downloader::startDownload(const std::string& uri, const Options& options)
+std::shared_ptr<Download> Downloader::startDownload(const std::string& uri, const std::string& downloadDir, Options options)
 {
   std::unique_lock<std::mutex> lock(m_lock);
+  for (auto it = options.begin(); it != options.end(); it++) {
+    if (it->first == "dir") {
+      it = options.erase(it);
+      break;
+    }
+  }
```

Comment:
Better remove *all* occurrences of "dir". Take advantage of [`std::remove_if()` STL algorithm](https://en.cppreference.com/w/cpp/algorithm/remove) (note that it doesn't remove the undesired elements, but moves them to the end, so you'll have to call `options.erase()` too).